stdlib/zipfile/__init__.pyi

```diff
@@ -225,8 +230,17 @@ class ZipFile:
     def getinfo(self, name: str) -> ZipInfo: ...
     def infolist(self) -> list[ZipInfo]: ...
     def namelist(self) -> list[str]: ...
+    @overload
+    def open(
+        self, name: str | ZipInfo, mode: Literal["r"] = "r", pwd: bytes | None = None, *, force_zip64: bool = False
+    ) -> ZipExtFile: ...
+    @overload
+    def open(
+        self, name: str | ZipInfo, mode: Literal["w"] = ..., pwd: bytes | None = None, *, force_zip64: bool = False
+    ) -> _ZipWriteFile: ...
+    @overload
     def open(
-        self, name: str | ZipInfo, mode: _ReadWriteMode = "r", pwd: bytes | None = None, *, force_zip64: bool = False
+        self, name: str | ZipInfo, mode: _ReadWriteMode, pwd: bytes | None = None, *, force_zip64: bool = False
     ) -> IO[bytes]: ...
```

Comment:
I suggest to leave out this "fallback" overload. If this causes too much trouble, mypy_primer in our CI should warn us.